src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -670,6 +702,73 @@ private int updateStopTimesForPatternStop(ObjectNode patternStop, int previousTr
         return travelTime + dwellTime;
     }
 
+    /**
+     * Normalizes all stop times' arrivals and departures for an ordered set of pattern stops. This set can be the full
+     * set of stops for a pattern or just a subset. Typical usage for this method would be to overwrite the arrival and
+     * departure times for existing trips after a pattern stop has been added or inserted into a pattern or if a
+     * pattern stop's default travel or dwell time were updated and the stop times need to reflect this update.
+     * @param patternStops list of pattern stops for which to update stop times (ordered by increasing stop_sequence)
+     * @param serviceIdFilters service_id values to filter trips on
+     * @throws SQLException
+     */
+    private int updateStopTimesForPatternStops(List<PatternStop> patternStops, Set<String> serviceIdFilters) throws SQLException {
+        PatternStop firstPatternStop = patternStops.iterator().next();
+        int firstStopSequence = firstPatternStop.stop_sequence;
+        // Prepare SQL query to determine the time that should form the basis for adding the travel time values.
+        int previousStopSequence = firstStopSequence > 0 ? firstStopSequence - 1 : 0;
+        String timeField = firstStopSequence > 0 ? "departure_time" : "arrival_time";
+        String getFirstTravelTimeSql = String.format(
+            "select t.trip_id, %s from %s.stop_times st, %s.trips t where stop_sequence = ? " +
+                "and t.pattern_id = ? " +
+                "and t.trip_id = st.trip_id",
+            timeField,
+            tablePrefix,
+            tablePrefix
+        );
+        if (serviceIdFilters != null && serviceIdFilters.size() > 0) {
+            getFirstTravelTimeSql += String.format(
+                " and t.service_id in (%s)",
+                Collections.nCopies(serviceIdFilters.size(), "?")
```

Comment:
Does this work? Wouldn't this just create a bunch of question marks without commas? It doesn't appear that this filter is ever used anyways. Is it needed?

Reply:
Also, there aren't any prepared statement set statements for these extra parameters either.